e2e/puppeteer-custom-environment.ts

```diff
@@ -31,12 +31,16 @@ class PuppeteerCustomEnvironment extends PuppeteerEnvironment {
     switch (event.name) {
       case 'test_fn_failure':
       case 'hook_failure':
-        const testParentName = state.currentlyRunningTest.parent.name.replace(/\W/g, '-');
-        const testName = state.currentlyRunningTest.name.replace(/\W/g, '-');
-        console.error(`Failed test:  "${testParentName}/${testName}"\n${event.error}`);
+        const runningTest = state.currentlyRunningTest;
+        let testName;
+        if (runningTest != null) {
+          testName = state.currentlyRunningTest.name.replace(/\W/g, '-');
+        } else {
+          testName = event.test.name.replace(/\W/g, '-');
+        }
 
-        const screenshotDir = `logs/screenshot/${testParentName}`;
-        const htmlDir = `logs/html/${testParentName}`;
+        const screenshotDir = `logs/screenshot`;
```

Comment:
Does this impact where some output files get sent (e.g. to a single top-level folder instead of a test-specific subfolder)? Is that something that will impact normal result viewing or test iteration? I'm wondering if any docs need to change as a result.

Reply:
yes, all test html/screenshots files save in same `log` folder.  It shouldn't affect anything else. `testParentName` is `null` on rare occasions and tests can fail if it is null. it's very annoying. No doc update for this change.